starter-code/pizza.js

```diff
@@ -17,69 +17,144 @@ var state = {
   greenPeppers: true,
   whiteSauce: false,
   glutenFreeCrust: false
-}
+};
 
 // This function takes care of rendering the pizza based on the state
 // This function is triggered once at the begining and everytime the state is changed
 function renderEverything() {
-  renderPepperonni()
-  renderMushrooms()
-  renderGreenPeppers()
-  renderWhiteSauce()
-  renderGlutenFreeCrust()
-
-  renderButtons()
-  renderPrice()
+  renderPepperonni();
+  renderMushrooms();
+  renderGreenPeppers();
+  renderWhiteSauce();
+  renderGlutenFreeCrust();
+  renderButtons();
+  renderPrice();
 }
 
 function renderPepperonni() {
-  document.querySelectorAll('.pep').forEach(function($pep){
+  document.querySelectorAll(".pep").forEach(function($pep) {
     if (state.pepperonni) {
       $pep.style.visibility = "visible";
-    }
-    else {
+    } else {
       $pep.style.visibility = "hidden";
     }
-  })
+  });
 }
 
 function renderMushrooms() {
-  // Iteration 1: set the visibility of `<section class="mushroom">`
+  document.querySelectorAll(".mushroom").forEach(function(mushrooms) {
+    if (state.mushrooms) {
+      mushrooms.style.visibility = "visible";
+    } else {
+      mushrooms.style.visibility = "hidden";
+    }
+  });
 }
 
 function renderGreenPeppers() {
-  // Iteration 1: set the visibility of `<section class="green-pepper">`
+  document.querySelectorAll(".green-pepper").forEach(function(peppers) {
+    if (state.greenPeppers) {
+      peppers.style.visibility = "visible";
+    } else {
+      peppers.style.visibility = "hidden";
+    }
+  });
 }
 
 function renderWhiteSauce() {
-  // Iteration 2: add/remove the class "sauce-white" of `<section class="sauce">`
+  document.querySelectorAll(".sauce-white").forEach(function(whiteSauce) {
+    if (state.whiteSauce) {
+      whiteSauce.style.visibility = "visible";
+    } else {
+      whiteSauce.style.visibility = "hidden";
+    }
+  });
 }
 
 function renderGlutenFreeCrust() {
-  // Iteration 2: add/remove the class "crust-gluten-free" of `<section class="crust">`
+  if (state.glutenFreeCrust) {
+    document.querySelector(".crust").classList.add("crust-gluten-free");
+  } else {
+    document.querySelector(".crust").classList.remove("crust-gluten-free");
+  }
 }
 
 function renderButtons() {
   // Iteration 3: add/remove the class "active" of each `<button class="btn">`
-}
 
-function renderPrice() {
-  // Iteration 4: change the HTML of `<aside class="panel price">`
```

Comment:
Hey Tim!
Just a piece of advice here. Whenever you feel like youre rewriting the same structure over and over again, theres probably something you can do to make it easier for yourself. Instead of all these if statements, you could have used the same logic as we showed for the prices:
```
function renderButtons() {
  for (const key in ingredientClasses) {
    if (state[key]) {
      document
        .querySelector(`.btn-${ingredientClasses[key]}`)
        .classList.add("active");
    } else
      document
        .querySelector(`.btn-${ingredientClasses[key]}`)
        .classList.remove("active");
  }
}
```

If this doesnt make sense let me know and we can go through it together ;)